skfem/assembly/basis/subdomain_facet_basis.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional, Tuple
+
+import numpy as np
+from numpy import ndarray
+from skfem.element import Element
+from skfem.mapping import Mapping
+from skfem.mesh import Mesh
+
+from .boundary_facet_basis import BoundaryFacetBasis
+from ..dofs import Dofs
+
+
+class SubdomainFacetBasis(BoundaryFacetBasis):
+
+    def __init__(self,
+                 mesh: Mesh,
+                 elem: Element,
+                 elements: ndarray,
+                 mapping: Optional[Mapping] = None,
+                 intorder: Optional[int] = None,
+                 quadrature: Optional[Tuple[ndarray, ndarray]] = None,
+                 side: int = 0,
+                 dofs: Optional[Dofs] = None):
+        # Used by .with_element()
+        self._elements = elements
+        self._side = side
+
+        elements = mesh.normalize_elements(elements)
+        # Get indices of all facets in the mesh
+        all_facets, all_counts = np.unique(mesh.t2f,
+                                           return_counts=True)
+        # Find the exterior ones
+        exterior_facets = all_facets[all_counts == 1]
```

Comment:
Isn't this same as `mesh.boundary_facets()`?

Reply:
If yes, you might want to change because something similar to the `np.unique` call is done already there.